server/models/plugins/cardCheckListPlugin.js

```diff
@@ -0,0 +1,59 @@
+const { Schema } = require("mongoose");
+const BasePlugin = require("./BasePlugin");
+
+const CheckListSchema = new Schema({
+  checklist: [{ item: String, completed: Boolean }],
+});
+
+const CardChecklistPluginSchema = BasePlugin.discriminator(
+  "CardCheckListPlugin",
+  new Schema(
+    {
+      cardCheckList: [CheckListSchema],
+    },
+    { timestamps: true }
+  )
+);
+
+CardChecklistPluginSchema.methods.get = async (pluginId) => {
+  const plugin = await this.findById(pluginId);
+  return plugin.checklist;
+};
+
+CardChecklistPluginSchema.methods.create = async (pluginId, checklist) => {
+  const plugin = await this.findById(pluginId);
+  if (plugin.checklist.item.length === 0) {
+    plugin.checklist = checklist;
+    await plugin.save();
+  } else {
+    throw new Error("Cannot rewrite files.");
+  }
+};
+
+CardChecklistPluginSchema.methods.update = async (pluginId, checklist) => {
+  const plugin = await this.findbyId(pluginId);
+  checklist.map((list) => {
+    return plugin.checklist.push(list);
+  });
+  await plugin.save();
+};
+
+CardChecklistPluginSchema.methods.patch = async (pluginId) => {
+  const plugin = await this.findById(pluginId);
+  if (plugin.checklist.completed === false) {
+    plugin.checklist.completed = true;
+    await plugin.save();
+  } else {
+    plugin.checklist.completed = false;
+    await plugin.save();
+  }
+};
```

Comment:
This code doesn't allow for a checklist item to be marked as complete, use query params to differentiate between different items.